From dc3a40e02cfa87a9667ac3d3525fa421e9b69728 Mon Sep 17 00:00:00 2001
From: Minecrell <minecrell@minecrell.net>
Date: Mon, 21 Sep 2020 11:18:26 +0200
Subject: [PATCH 48/78] arm64: dts: qcom: msm8916: Add common
 msm8916-modem.dtsi (v3)

There are quite a lot of status switches we need to flip, so let's
add a common device tree include to simplify enabling all modem
functionality.

v2: - Add BAM DMUX as network interface to the modem
    - Remove qcom,audio-routing
v3: Add memshare for GPS

Co-Authored-By: Nikita Travkin <nikitos.tr@gmail.com>
---
 arch/arm64/boot/dts/qcom/msm8916-modem.dtsi | 91 +++++++++++++++++++++
 1 file changed, 91 insertions(+)
 create mode 100644 arch/arm64/boot/dts/qcom/msm8916-modem.dtsi

diff --git a/arch/arm64/boot/dts/qcom/msm8916-modem.dtsi b/arch/arm64/boot/dts/qcom/msm8916-modem.dtsi
new file mode 100644
index 000000000..fe1d2db14
--- /dev/null
+++ b/arch/arm64/boot/dts/qcom/msm8916-modem.dtsi
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <dt-bindings/sound/qcom,q6afe.h>
+#include <dt-bindings/sound/qcom,q6asm.h>
+#include <dt-bindings/sound/qcom,q6voice.h>
+
+&apr {
+	status = "okay";
+};
+
+&bam_dmux {
+	status = "okay";
+};
+
+&bam_dmux_dma {
+	status = "okay";
+};
+
+&gps_mem {
+	status = "okay";
+};
+
+&lpass {
+	status = "disabled";
+};
+
+&memshare {
+	status = "okay";
+};
+
+&memshare_gps {
+	memory-region = <&gps_mem>;
+};
+
+&mpss {
+	status = "okay";
+};
+
+&mpss_mem {
+	status = "okay";
+};
+
+&q6cvp {
+	status = "okay";
+};
+
+&q6cvs {
+	status = "okay";
+};
+
+&q6mvm {
+	status = "okay";
+};
+
+&sound {
+	compatible = "qcom,msm8916-qdsp6-sndcard";
+
+	dai-link-multimedia1 {
+		link-name = "MultiMedia1";
+		cpu {
+			sound-dai = <&q6asmdai MSM_FRONTEND_DAI_MULTIMEDIA1>;
+		};
+	};
+
+	dai-link-multimedia2 {
+		link-name = "MultiMedia2";
+		cpu {
+			sound-dai = <&q6asmdai MSM_FRONTEND_DAI_MULTIMEDIA2>;
+		};
+	};
+
+	dai-link-multimedia3 {
+		link-name = "MultiMedia3";
+		cpu {
+			sound-dai = <&q6asmdai MSM_FRONTEND_DAI_MULTIMEDIA3>;
+		};
+	};
+
+	dai-link-multimedia4 {
+		link-name = "MultiMedia4";
+		cpu {
+			sound-dai = <&q6asmdai MSM_FRONTEND_DAI_MULTIMEDIA4>;
+		};
+	};
+
+	dai-link-cs-voice {
+		link-name = "CS-Voice";
+		cpu {
+			sound-dai = <&q6voicedai CS_VOICE>;
+		};
+	};
+};
-- 
2.31.1

